rmgpy/molecule/graph.pyx

```diff
@@ -972,10 +972,29 @@ cdef class Graph:
 
         rc = []
         for cycle in data.get_rcs():
-            rc.append([graph.get_node_for_index(i) for i in cycle.nodes])
+            rc.append(self._sortCyclicVertices([graph.get_node_for_index(i) for i in cycle.nodes]))
 
         return rc
 
+    cpdef list _sortCyclicVertices(self, list vertices):
+        """
+        Given a list of vertices comprising a cycle, sort them such that adjacent
+        entries in the list are connected to each other.
+        Warning: Assumes that the cycle is elementary, ie. no bridges.
+        """
+        cdef list ordered
+        cdef Vertex vertex
+
+        ordered = [vertices.pop()]
+        while vertices:
+            for vertex in vertices:
+                if vertex in ordered[-1].edges:
+                    ordered.append(vertex)
+                    vertices.remove(vertex)
+                    break
+
```

Comment:
Currently the method accept any list of vertices, if one vertex that is out of a cyclic is accidentally put in, it seems easily to end up in an infinite loop

Reply:
Thanks, I'll do that.